### ISSUE TITLE
Cannot Update; Throwing Error "No module named 'distutils.command.bdist_wininst' "

### ISSUE THREAD

**Issue by AgentLoneStar007** [openrazer/razer-drivers#1972]
### Distribution
EndeavourOS(Arch-based)

Kernel version 6.0.10-arch2-1

### Support

- [ ] Have you tried rebooting?
- [x] Have you looked at the [Troubleshooting](https://github.com/openrazer/openrazer/wiki/Troubleshooting) page?

Hi. I simply tried to update everything this morning on my PC using `yay -Syu`, and everything went well until it tried to compile the OpenRazer Daemon, which resulted in a Python error of "`No module named 'distutils.command.bdist_wininst'`".
Full output log:
```
:: Installing OpenRazer python library
=====================================================
python3 setup.py install --prefix=/usr --root=/home/agent/.cache/yay/openrazer/pkg/python-openrazer
/usr/lib/python3.10/site-packages/setuptools_scm/integration.py:27: RuntimeWarning: 
ERROR: setuptools==44.1.1 is used in combination with setuptools_scm>=6.x

Your build configuration is incomplete and previously worked by accident!
setuptools_scm requires setuptools>=45


This happens as setuptools is unable to replace itself when a activated build dependency
requires a more recent setuptools version
(it does not respect "setuptools>X" in setup_requires).


setuptools>=31 is required for setup.cfg metadata support
setuptools>=42 is required for pyproject.toml configuration support

Suggested workarounds if applicable:
 - preinstalling build dependencies like setuptools_scm before running setup.py
 - installing setuptools_scm using the system package manager to ensure consistency
 - migrating from the deprecated setup_requires mechanism to pep517/518
   and using a pyproject.toml to declare build dependencies
   which are reliably pre-installed before running the build tools

  warnings.warn(
running install
running build
running build_py
creating build
creating build/lib
creating build/lib/openrazer
copying openrazer/__init__.py -> build/lib/openrazer
creating build/lib/openrazer/client
copying openrazer/client/macro.py -> build/lib/openrazer/client
copying openrazer/client/device.py -> build/lib/openrazer/client
copying openrazer/client/fx.py -> build/lib/openrazer/client
copying openrazer/client/__init__.py -> build/lib/openrazer/client
copying openrazer/client/constants.py -> build/lib/openrazer/client
copying openrazer/client/debug.py -> build/lib/openrazer/client
creating build/lib/openrazer/client/devices
copying openrazer/client/devices/mousemat.py -> build/lib/openrazer/client/devices
copying openrazer/client/devices/__init__.py -> build/lib/openrazer/client/devices
copying openrazer/client/devices/mice.py -> build/lib/openrazer/client/devices
copying openrazer/client/devices/keyboard.py -> build/lib/openrazer/client/devices
running install_lib
creating /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr
creating /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib
creating /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10
creating /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages
creating /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer
creating /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client
copying build/lib/openrazer/client/macro.py -> /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client
copying build/lib/openrazer/client/device.py -> /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client
copying build/lib/openrazer/client/fx.py -> /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client
copying build/lib/openrazer/client/__init__.py -> /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client
copying build/lib/openrazer/client/constants.py -> /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client
copying build/lib/openrazer/client/debug.py -> /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client
creating /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/devices
copying build/lib/openrazer/client/devices/mousemat.py -> /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/devices
copying build/lib/openrazer/client/devices/__init__.py -> /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/devices
copying build/lib/openrazer/client/devices/mice.py -> /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/devices
copying build/lib/openrazer/client/devices/keyboard.py -> /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/devices
copying build/lib/openrazer/__init__.py -> /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer
byte-compiling /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/macro.py to macro.cpython-310.pyc
byte-compiling /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/device.py to device.cpython-310.pyc
byte-compiling /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/fx.py to fx.cpython-310.pyc
byte-compiling /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/__init__.py to __init__.cpython-310.pyc
byte-compiling /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/constants.py to constants.cpython-310.pyc
byte-compiling /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/debug.py to debug.cpython-310.pyc
byte-compiling /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/devices/mousemat.py to mousemat.cpython-310.pyc
byte-compiling /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/devices/__init__.py to __init__.cpython-310.pyc
byte-compiling /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/devices/mice.py to mice.cpython-310.pyc
byte-compiling /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/client/devices/keyboard.py to keyboard.cpython-310.pyc
byte-compiling /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer/__init__.py to __init__.cpython-310.pyc
running install_egg_info
running egg_info
creating openrazer.egg-info
writing openrazer.egg-info/PKG-INFO
writing dependency_links to openrazer.egg-info/dependency_links.txt
writing requirements to openrazer.egg-info/requires.txt
writing top-level names to openrazer.egg-info/top_level.txt
writing manifest file 'openrazer.egg-info/SOURCES.txt'
reading manifest file 'openrazer.egg-info/SOURCES.txt'
writing manifest file 'openrazer.egg-info/SOURCES.txt'
Copying openrazer.egg-info to /home/agent/.cache/yay/openrazer/pkg/python-openrazer/usr/lib/python3.10/site-packages/openrazer-3.5.1-py3.10.egg-info
running install_scripts
Traceback (most recent call last):
  File "/home/agent/.cache/yay/openrazer/src/openrazer-3.5.1/pylib/setup.py", line 6, in <module>
    setup(
  File "/home/agent/.local/lib/python3.10/site-packages/setuptools/__init__.py", line 162, in setup
    return distutils.core.setup(**attrs)
  File "/usr/lib/python3.10/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/usr/lib/python3.10/distutils/dist.py", line 966, in run_commands
    self.run_command(cmd)
  File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/agent/.local/lib/python3.10/site-packages/setuptools/command/install.py", line 61, in run
    return orig.install.run(self)
  File "/usr/lib/python3.10/distutils/command/install.py", line 580, in run
    self.run_command(cmd_name)
  File "/usr/lib/python3.10/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/usr/lib/python3.10/distutils/dist.py", line 985, in run_command
    cmd_obj.run()
  File "/home/agent/.local/lib/python3.10/site-packages/setuptools/command/install_scripts.py", line 35, in run
    bw_cmd = self.get_finalized_command("bdist_wininst")
  File "/usr/lib/python3.10/distutils/cmd.py", line 298, in get_finalized_command
    cmd_obj = self.distribution.get_command_obj(command, create)
  File "/usr/lib/python3.10/distutils/dist.py", line 857, in get_command_obj
    klass = self.get_command_class(command)
  File "/home/agent/.local/lib/python3.10/site-packages/setuptools/dist.py", line 834, in get_command_class
    self.cmdclass[command] = cmdclass = ep.load()
  File "/home/agent/.local/lib/python3.10/site-packages/pkg_resources/__init__.py", line 2443, in load
    return self.resolve()
  File "/home/agent/.local/lib/python3.10/site-packages/pkg_resources/__init__.py", line 2449, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/home/agent/.local/lib/python3.10/site-packages/setuptools/command/bdist_wininst.py", line 1, in <module>
    import distutils.command.bdist_wininst as orig
ModuleNotFoundError: No module named 'distutils.command.bdist_wininst'
make[1]: *** [Makefile:9: install] Error 1
make: *** [Makefile:91: python_library_install] Error 2
==> ERROR: A failure occurred in package_python-openrazer().
    Aborting...
```

**Comment by z3ntu** [comment]
Try building in a clean chroot, e.g. using `extra-x86_64-build` https://wiki.archlinux.org/title/DeveloperWiki:Building_in_a_clean_chroot

**Comment by z3ntu** [comment]
Also Arch repos contain setuptools v65, not v44 as you seem to have installed according to the error message.
```
$ pacman -Q python-setuptools
python-setuptools 1:65.6.3-1
```
Are you running something like `sudo pip install` sometimes?

**Comment by AgentLoneStar007** [comment]
I've never ran pip in sudo before.

**Comment by AgentLoneStar007** [comment]
Alright, so after using a the devtools script in the package directory, it gave this error at the end:
```
checking keyring...
checking package integrity...
loading package files...
checking for file conflicts...
:: Processing package changes...
installing pyalpm...
installing licenses...
installing elfutils...
installing python-pyelftools...
installing namcap...
:: Running post-transaction hooks...
(1/1) Arming ConditionNeedsUpdate...
Checking PKGBUILD
Checking openrazer-daemon-3.5.1-1-any.pkg.tar.zst
openrazer-daemon W: Dependency python-dbus included but already satisfied
Checking openrazer-driver-dkms-3.5.1-1-any.pkg.tar.zst
openrazer-driver-dkms W: Dependency included and not needed ('dkms')
openrazer-driver-dkms W: Dependency included and not needed ('udev')
Checking openrazer-meta-3.5.1-1-any.pkg.tar.zst
openrazer-meta W: Dependency included and not needed ('openrazer-driver-dkms')
openrazer-meta W: Dependency included and not needed ('openrazer-daemon')
openrazer-meta W: Dependency included and not needed ('python-openrazer')
Checking python-openrazer-3.5.1-1-any.pkg.tar.zst
==> Running checkpkg
error: target not found: python-openrazer
error: target not found: openrazer-daemon
error: target not found: openrazer-driver-dkms
error: target not found: openrazer-meta
==> WARNING: Skipped checkpkg due to missing repo packages
```
[Here's the full output.](https://pastebin.com/TLiGYZB2) (It's a bit long. Output of installing Openrazer and the build script.)

**Comment by z3ntu** [comment]
Looks good, build succeeded. Now you can install the *.pkg.tar.zst files produced by that. The checkpkg is basically just useful for actual Arch Linux developers that push packages to the repos.

**Comment by AgentLoneStar007** [comment]
Well, that worked. Thanks. It's funny, though, because _as_ I go to update to check, Protontricks is doing the exact same thing, and what I did for this didn't work on that.

Speaking of which, if anyone has this issue and needs help, here's what I did:

- Install `devtools` from the Arch repository
- Go to `~/.cache/yay/openrazer`
- Delete everything in that directory with `sudo rm -dr *`
- Run the command `wget https://aur.archlinux.org/cgit/aur.git/plain/PKGBUILD?h=openrazer && mv PKGBUILD?h=openrazer PKGBUILD` (Recreating the PKGBUILD file isn't necessary to my knowledge, but I like to do it.)
- Run the command `extra-x86_64-build` in that directory
- When finished, install all packages ending in `.pkg.tar.zst` with `sudo pacman -U (packagename)`
